lib/livebook_web/live/hub/edit/fly_component.ex

```diff
@@ -275,47 +151,60 @@ defmodule LivebookWeb.Hub.Edit.FlyComponent do
     end
   end
 
-  def handle_event("save", %{"env_var" => params}, socket) do
-    if socket.assigns.valid_env_var? do
-      case FlyClient.put_secrets(socket.assigns.hub, [params]) do
-        {:ok, _} ->
-          message =
-            if socket.assigns.operation == :new do
-              "Environment variable added"
-            else
-              "Environment variable updated"
-            end
+  def handle_event("validate", %{"fly" => attrs}, socket) do
+    {:noreply, assign(socket, changeset: Fly.change_hub(socket.assigns.hub, attrs))}
+  end
 
-          {:noreply,
-           socket
-           |> put_flash(:success, message)
-           |> push_redirect(to: Routes.hub_path(socket, :edit, socket.assigns.hub.id))}
+  # EnvVar component callbacks
 
-        {:error, _} ->
-          message =
-            if socket.assigns.operation == :new do
-              "Failed to add environment variable"
-            else
-              "Failed to update environment variable"
-            end
+  def handle_event("save_env_var", %{"env_var" => attrs}, socket) do
+    case FlyClient.put_secrets(socket.assigns.hub, [attrs]) do
+      {:ok, _} ->
+        message =
+          if socket.assigns.env_operation == :new do
```

Comment:
I would make the modal send the information if it was a "new" or "edit". This way we don't need to keep track of `env_operation` here.